operations/mimir-mixin/docs/playbooks.md

```diff
@@ -1011,6 +1011,99 @@ How to **investigate**:
+
+> **Note**: Invalid series are skipped during the ingestion, while other valid series within the same request are ingested.
+
+### err-mimir-duplicate-label-names
+
+This non-critical error occurs when Mimir receives a write request containing a series with the same label name repeated two or more times.
+A series containing repeated label names is invalid and so it gets skipped during the ingestion.
+
+> **Note**: Invalid series are skipped during the ingestion, while other valid series within the same request are ingested.
+
+### err-mimir-labels-not-sorted
+
+This error occurs when Mimir receives a write request containing a series whose label names are not sorted alphabetically.
+However, Mimir internally sort labels for received series, so this error shouldn't occur in pratice.
+If you experience this error, please open an issue in the Mimir repository.
+
+> **Note**: Invalid series are skipped during the ingestion, while other valid series within the same request are ingested.
+
+### err-mimir-too-far-in-future
+
+This non-critical error occurs when Mimir receives a write request containing a sample whose timestamp is in the future compared to the current "real world" time.
+Mimir tolerates timestamps slightly in the future (e.g. due to clock screws), while rejects timestamps more then `-validation.create-grace-period` in the future.
+You can fine tune the toleration on a per-tenant basis using `-validation.max-length-label-value`.
+
+> **Note**: Series with invalid samples are skipped during the ingestion, while other valid series within the same request are ingested.
+
+### err-mimir-exemplar-labels-missing
+
+This non-critical error occurs when Mimir receives a write request containing an exemplar without any label identifying the related metric.
+An exemplar must have at least a valid label pairs, otherwise it can't be associated to any metric.
+
+> **Note**: Invalid exemplars skipped during the ingestion, while other valid exemplars within the same request are ingested.
+
+### err-mimir-exemplar-labels-too-long
+
+This non-critical error occurs when Mimir receives a write request containing an exemplar whose combined labels set size exceeds the limit.
+The limit is used to protect the system stability from potential abuses or mistakes, and it can't be configured.
+
+> **Note**: Invalid exemplars skipped during the ingestion, while other valid exemplars within the same request are ingested.
+
+### err-mimir-exemplar-timestamp-invalid
+
+This non-critical error occurs when Mimir receives a write request containing an exemplar without timestamp.
+An exemplar must have a valid timestamp, otherwise it can't be correlated to any point in time.
+
+> **Note**: Invalid exemplars skipped during the ingestion, while other valid exemplars within the same request are ingested.
```

Comment:
```suggestion
### err-mimir-label-invalid

This non-critical error occurs when Mimir receives a write request that contains a series with an invalid label name.
A label name name can only contain characters as defined by Prometheus’ [Metric names and labels](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels).

> **Note**: Invalid series are skipped during the ingestion, and valid series within the same request are ingested.

### err-mimir-label-name-too-long

This non-critical error occurs when Mimir receives a write request that contains a series with a label name whose length exceeds the configured limit.
The limit protects the system’s stability from potential abuse or mistakes, and you can configure the limit on a per-tenant basis by using the `-validation.max-length-label-name` option.

> **Note**: Invalid series are skipped during the ingestion, and valid series within the same request are ingested.

### err-mimir-label-value-too-long

This non-critical error occurs when Mimir receives a write request that contains a series with a label value whose length exceeds the configured limit.
The limit protects the system’s stability from potential abuse or mistakes, and you can configure the limit on a per-tenant basis by using the `-validation.max-length-label-value` option.

> **Note**: Invalid series are skipped during the ingestion, and valid series within the same request are ingested.

### err-mimir-duplicate-label-names

This non-critical error occurs when Mimir receives a write request that contains a series with the same label name two or more times.
A series that contains a duplicated label name is invalid and gets skipped during the ingestion.

> **Note**: Invalid series are skipped during the ingestion, and valid series within the same request are ingested.

### err-mimir-labels-not-sorted

This error occurs when Mimir receives a write request that contains a series whose label names are not sorted alphabetically.
However, Mimir internally sorts labels for series that it receives, so this error should not occur in practice.
If you experience this error, [open an issue in the Mimir repository](https://github.com/grafana/mimir/issues).

> **Note**: Invalid series are skipped during the ingestion, and valid series within the same request are ingested.

### err-mimir-too-far-in-future

This non-critical error occurs when Mimir receives a write request that contains a sample whose timestamp is in the future compared to the current "real world" time.
Mimir accepts timestamps that are slightly in the future, due to skewed clocks for example. It rejects timestamps that are too far in the future, based on the definition that you can set via the `-validation.create-grace-period` option.
On a per-tenant basis, you can fine tune the tolerance by configuring the `-validation.max-length-label-value` option.

> **Note**: Series with invalid samples are skipped during the ingestion, and series within the same request are ingested.

### err-mimir-exemplar-labels-missing

This non-critical error occurs when Mimir receives a write request that contains an exemplar without a label that identifies the related metric.
An exemplar must have at least one valid label pair, otherwise it cannot be associated with any metric.

> **Note**: Invalid exemplars are skipped during the ingestion, and valid exemplars within the same request are ingested.

### err-mimir-exemplar-labels-too-long

This non-critical error occurs when Mimir receives a write request that contains an exemplar where the combined set size of its labels exceeds the limit.
The limit is used to protect the system’s stability from potential abuse or mistakes, and it cannot be configured.

> **Note**: Invalid exemplars are skipped during the ingestion, and valid exemplars within the same request are ingested.

### err-mimir-exemplar-timestamp-invalid

This non-critical error occurs when Mimir receives a write request that contains an exemplar without a timestamp.
An exemplar must have a valid timestamp, otherwise it cannot be correlated to any point in time.

> **Note**: Invalid exemplars are skipped during the ingestion, and valid exemplars within the same request are ingested.
```

Reply:
Well, that makes reviewing difficult. I made similar changes to those previous to these lines. I will make a comment about the ones that deviate from repetition to make the review easier.